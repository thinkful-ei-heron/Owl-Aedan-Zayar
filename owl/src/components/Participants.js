import React from 'react';
import './Participants.css';

export default class Participants extends React.Component {
    

    render() {
        let parti = this.props.part.map(item => {
            return (
            <li key={item.id} className = 'participants'>
                <img src={item.avatar} alt="Avatar"/>
                <div>
                    <p>{item.name}</p>
                    <span className ={`inSession${item.inSession}`}/>
                    <span className ={`onStage${item.onStage}`}/>
                </div>
            </li>);
        });
        return (
        <ul>
           {parti}
        </ul> 
        );
    }
}
