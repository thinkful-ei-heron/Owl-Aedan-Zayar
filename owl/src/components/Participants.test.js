import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participants from './Participants';
import store from '../store';

describe('Participants component', () => {
    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participants part={store.participants}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Renders the UI as expected', () => {
        const tree = renderer
        .create(<Participants part={store.participants}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
