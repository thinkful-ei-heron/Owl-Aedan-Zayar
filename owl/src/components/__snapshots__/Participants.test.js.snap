// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants component Renders the UI as expected 1`] = `
<ul>
  <li>
    <img
      alt="Avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      <p>
        Koren Templeton
      </p>
      <span
        className="inSessiontrue"
      />
      <span
        className="onStagetrue"
      />
    </div>
  </li>
  <li>
    <img
      alt="Avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      <p>
        Caty Flucker
      </p>
      <span
        className="inSessiontrue"
      />
      <span
        className="onStagefalse"
      />
    </div>
  </li>
  <li>
    <img
      alt="Avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      <p>
        Axe Kubicka
      </p>
      <span
        className="inSessionfalse"
      />
      <span
        className="onStagefalse"
      />
    </div>
  </li>
  <li>
    <img
      alt="Avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      <p>
        Frank Runciman
      </p>
      <span
        className="inSessiontrue"
      />
      <span
        className="onStagefalse"
      />
    </div>
  </li>
  <li>
    <img
      alt="Avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      <p>
        Ashla Attwool
      </p>
      <span
        className="inSessiontrue"
      />
      <span
        className="onStagetrue"
      />
    </div>
  </li>
</ul>
`;
